xdsl/dialects/vector.py

```diff
@@ -169,4 +169,76 @@ def get(memref: SSAValue | Operation, indices: List[SSAValue | Operation],
                                 ])
 
 
-Vector = Dialect([Load, Store, Broadcast, FMA, Maskedload], [])
+@irdl_op_definition
+class Maskedstore(Operation):
+    name = "vector.maskedstore"
+    memref: Annotated[Operand, MemRefType]
+    indices: Annotated[VarOperand, IndexType]
+    mask: Annotated[Operand, VectorType]
+    value_to_store: Annotated[Operand, VectorType]
+
+    def verify_(self):
+        memref_element_type = self.memref.typ.element_type
+
+        if memref_element_type != self.value_to_store.typ.element_type:
+            raise VerifyException(
+                "MemRef element type should match the stored vector type. Obtained types were "
+                + str(memref_element_type) + " and " +
+                str(self.value_to_store.typ.element_type) + ".")
+
+        if len(self.value_to_store.typ.get_shape()) != 1:
+            raise VerifyException("Expected a rank 1 vector to be stored.")
+
+        if len(self.mask.typ.get_shape()) != 1:
+            raise VerifyException("Expected a rank 1 mask vector.")
+
+        if self.mask.typ.element_type != i1:
```

Comment:
IMO, it would be cool to be able to encode this invariant as part of the annotation instead of in the verifier. That would make our annotation more declarative instead of the imperative verifier here.Though this might be more of a topic for a later, bigger xDSL PR. (@math-fehr)

Reply:
Yes, I agree. I was thinking of modelling at least `VectorOfRankAndType` (for verifying vector rank and type) and `VectorOfRank` (for verifying vector rank) verifier annotations for the `vector` dialect. I'll have a look at other dialects to see if they have repeated verifier mechanisms as well. Perhaps we should address all of them across all dialects in a separate PR.